macros/core/scaffold.pl

```diff
@@ -676,7 +641,7 @@ sub can_open {
 	my $method = ($Scaffold::isInstructor ? $self->{instructor_can_open} : $self->{can_open});
 	$method = $self->{after_AnswerDate_can_open} if $Scaffold::afterAnswerDate;
 	$method = "Section::can_open::" . $method unless ref($method) eq 'CODE';
-	return &{$method}($self);
+	return $self->$method;
```

Comment:
Note that `$method` can be either a method name in the `Section::can_open` package (the usual case), or a CODE literal.  (See the POD documentation for `can_open`, and note the `unless ref($method) eq 'CODE'` in the line above).

You probably need to do something like

``` js
if (ref($method) eq 'CODE') {
  return &{$method}($self);
}
$method = "Section::can_open::";
return $self->$method;
```

instead in order to handle both cases.  Similarly for the `is_open` code below.

Reply:
Yeah, I forgot that possibility.  I have fixed that now.  Thanks.